### ISSUE TITLE
✅ SOLVED - 🐞Global Widgets are not appearing on the frontend when used inside Theme Builder templates

### ISSUE THREAD

**Issue by LoganStrike** [elementor/elementor#29092]
### Prerequisites

- [X] I have searched for similar issues in open and closed tickets and cannot find a duplicate.
- [X] I have troubleshooted my issue, and it still exists against the latest stable version of Elementor.

### Description

After updating to Elementor 3.25.1, when editing a Theme Builder template, all Global modules that were previously placed on the template are no longer on that template. The global elements are visible on front end, but missing in the Elementor editor.

### Steps to reproduce

1. Updated a site to Elementor 3.25.1.
2. Edit a theme builder template that contains Global elements. The Global elements are not there. The parent column is empty.

Note that on this particular website, it uses the previous grid system with Sections/Inner Section instead of Container.

Also tried clearing Elementor cache, server cache, and browser cache.

### Expected behavior

The global elmements should not be missing.

### Isolating the problem

- [X] This bug happens when only the Elementor (and Elementor Pro) plugins are active.
- [X] This bug happens with the Hello Elementor theme active.
- [X] I can reproduce this bug consistently by following the steps I described above.

### Elementor System Info

<details>
<summary>Click to reveal</summary>

```txt

== Server Environment ==
	Operating System: Linux
	Software: Flywheel/5.1.0
	MySQL version: (Google) v8.0.31
	PHP Version: 8.2.24
	PHP Memory Limit: 128M
	PHP Max Input Vars: 10000
	PHP Max Post Size: 300M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.6.2
	Site URL: ***
	Home URL: ***
	WP Multisite: No
	Max Upload Size: 300 MB
	Memory limit: 128M
	Max Memory limit: 128M
	Permalink Structure: /%category%/%postname%/
	Language: en_US
	Timezone: America/Phoenix
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elementor Child
	Version: 2.0.0
	Author: Elementor Team
	Child Theme: Yes
	Parent Theme Name: Hello Elementor
	Parent Theme Version: 3.1.1
	Parent Theme Author: Elementor Team

== User ==
	Role: administrator
	WP Profile lang: en-US
	User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/130.0.0.0 Safari/537.36

== Active Plugins ==
	Elementor
		Version: 3.25.1
		Author: Elementor.com

	Elementor Pro
		Version: 3.25.0
		Author: Elementor.com


== Features ==
	Custom Fonts: 0
	Custom Icons: 0

== Integrations ==
	


== Elementor Experiments ==
	Improved CSS Loading: Inactive by default
	Inline Font Icons: Inactive by default
	Additional Custom Breakpoints: Active by default
	Container: Inactive by default
	Upgrade Swiper Library: Active by default
	Nested Elements Performance: Active by default
	Optimized Control Loading: Active by default
	Optimized Markup: Inactive by default
	Conditionally load Swiper CSS files: Inactive by default
	Plugin Onboarding: Active by default
	CSS Smooth Scroll: Active by default
	Default to New Theme Builder: Active by default
	Header & Footer: Inactive by default
	Elementor Home Screen: Active by default
	Landing Pages: Active by default
	Nested Elements: Active by default
	Editor Top Bar: Active by default
	Pages Panel: Inactive by default
	Element Caching: Inactive by default
	Link In Bio: Active by default
	Floating Buttons: Active by default
	Elementor Editor Events: Inactive by default
	Atomic Widgets: Inactive by default
	Launchpad Checklist: Inactive by default
	Menu: Inactive by default


== Log ==
	
PHP: showing 7 of 7PHP: 2024-09-04 14:25:11 [notice X 1][/www/wp-content/plugins/elementor/modules/history/revisions-manager.php::363] Undefined index: editor_post_id [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2024-09-04 16:04:06 [notice X 2][/www/wp-content/plugins/elementor-pro/modules/dynamic-tags/acf/tags/acf-text.php::33] Undefined offset: 1 [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2024-10-11 15:58:33 [notice X 1][/www/wp-content/plugins/elementor/core/files/css/base.php::783] Undefined index: padding_mobile [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2024-10-12 19:09:46 [warning X 1][/www/wp-content/plugins/elementor-pro/modules/assets-manager/asset-types/icons/custom-icons.php::199] unlink(zillow-solid.svg): No such file or directory [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2024-10-12 19:27:32 [notice X 4][/www/wp-content/plugins/elementor-pro/modules/screenshots/module.php::33] Function utf8_decode() is deprecated [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2024-10-14 10:29:32 [warning X 1][/www/wp-content/plugins/elementor/includes/managers/image.php::113] Trying to access array offset on value of type bool [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2024-10-28 15:37:11 [error X 2][/www/wp-content/plugins/elementor-pro/modules/global-widget/data/controller.php::31] Uncaught Error: Cannot use object of type WP_Error as array in /www/wp-content/plugins/elementor-pro/modules/global-widget/data/controller.php:31
Stack trace:
#0 /wordpress/wp-includes/rest-api/class-wp-rest-server.php(1230): ElementorPro\Modules\GlobalWidget\Data\Controller->get_items(Object(WP_REST_Request))
#1 /wordpress/wp-includes/rest-api/class-wp-rest-server.php(1063): WP_REST_Server->respond_to_request(Object(WP_REST_Request), '/elementor/v1/g...', Array, NULL)
#2 /wordpress/wp-includes/rest-api/class-wp-rest-server.php(439): WP_REST_Server->dispatch(Object(WP_REST_Request))
#3 /wordpress/wp-includes/rest-api.php(420): WP_REST_Server->serve_request('/elementor/v1/g...')
#4 /wordpress/wp-includes/class-wp-hook.php(324): rest_api_loaded(Object(WP))
#5 /wordpress/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters('', Array)
#6 /wordpress/wp-includes/plugin.php(565): WP_Hook->do_action(Array)
#7 /wordpress/wp-includes/class-wp.php(418): do_action_ref_array('parse_request', Array)
#8 /wordpress/wp-includes/class-wp.php(813): WP->parse_request('')
#9 /wordpress/wp-includes/functions.php(1336): WP->main('')
#10 /wordpress/wp-blog-header.php(16): wp()
#11 /www/index.php(17): require('/wordpress/wp-b...')
#12 {main}
  thrown [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]

Log: showing 20 of 302024-10-22 15:15:19 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.24.6',
  'to' => '3.24.7',
)]
2024-10-22 15:15:19 [info] elementor::elementor_updater Started 
2024-10-22 15:15:20 [info] Elementor/Upgrades - _on_each_version Start  
2024-10-22 15:15:20 [info] Elementor/Upgrades - _on_each_version Finished 
2024-10-22 15:15:20 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.24.6',
  'to' => '3.24.7',
)]
2024-10-28 15:18:44 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.24.7',
  'to' => '3.25.0',
)]
2024-10-28 15:18:45 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.24.4',
  'to' => '3.25.0',
)]
2024-10-28 15:18:47 [info] elementor-pro::elementor_pro_updater Started 
2024-10-28 15:18:47 [info] elementor::elementor_updater Started 
2024-10-28 15:18:47 [info] Elementor Pro/Upgrades - _on_each_version Start  
2024-10-28 15:18:47 [info] Elementor/Upgrades - _on_each_version Start  
2024-10-28 15:18:47 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2024-10-28 15:18:47 [info] Elementor/Upgrades - _on_each_version Finished 
2024-10-28 15:18:47 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.24.4',
  'to' => '3.25.0',
)]
2024-10-28 15:58:25 [info] elementor::elementor_updater Started 
2024-10-28 15:58:25 [info] Elementor/Upgrades - _on_each_version Start  
2024-10-28 15:58:25 [info] Elementor/Upgrades - _on_each_version Finished 
2024-10-28 15:58:25 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.25.0',
  'to' => '3.25.1',
)]
2024-10-28 15:58:25 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.25.0',
  'to' => '3.25.1',
)]
2024-10-28 15:58:26 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.25.0',
  'to' => '3.25.1',
)]

JS: showing 5 of 5JS: 2024-10-13 01:51:55 [error X 1][***/wp-admin/load-scripts.php?c=0&loadchunk_0=jquery-core,jquery-migrate,utils,wp-hooks&ver=6.6.2:2:28767] undefined is not an object (evaluating 'c[0].value') 
JS: 2024-10-13 02:06:45 [error X 1][***/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.24.6:3:816493] Cannot read properties of undefined (reading 'get') 
JS: 2024-10-13 02:18:42 [error X 1][***/wp-admin/load-scripts.php?c=0&loadchunk_0=jquery-core,jquery-migrate,utils,wp-hooks&ver=6.6.2:2:28760] Cannot read properties of undefined (reading 'value') 
JS: 2024-10-15 21:48:49 [error X 2][***/wp-admin/load-scripts.php?c=0&loadchunk_0=jquery-ui-draggable,react,react-dom&ver=6.6.2:13:112013] Cannot read properties of undefined (reading 'toString') 
JS: 2024-10-28 22:38:34 [error X 1][***/wp-content/plugins/elementor/assets/js/common-modules.min.js?ver=3.25.0:2:56144] Model.isValidChild() should be implemented, please provide 'isValidChild' functionality. 



== Elementor - Compatibility Tag ==
	
	Elementor Pro: Compatible

== Elementor Pro - Compatibility Tag ==

```
</details> 

### Agreement

- [X] I agree that my issue may be closed without action if it doesn't meet all the requirements.

**Comment by rvisima** [comment]
Same issue here, but our widgets are missing on both the front and back end. The only solution is re-adding the widgets in the editor AND reverting to 3.24.7. Simply adding the widgets back still doesn't make them appear on the front end under 3.25.1.

**Comment by metamindEdu** [comment]
I have the same issue. 3.25.1

**Comment by doronwolf** [comment]
Hi :) Just writing to let you know we are aware of the issue and working on a fix

**Comment by Panark** [comment]
Elementor 3.25.1 and pro 3.25 broke multiple websites (visually).
Reverting is the only way to keep thing running at this time!!!

**Comment by Whitecitys** [comment]
I have the same issue

**Comment by dev12-byte** [comment]
+1 I have global templates that act as a spacer (Template with a Spacer inside) which is used on almost all pages. After updating they are missing from the page entirely, even in the editor!

UpdraftPlus was my savior.

**Comment by giacomocelli** [comment]
I have the same issue

**Comment by BigLeDouski** [comment]
I have the same isssue

**Comment by marianne90** [comment]
I have the same problem. Also contact forms are missing styling. The fields are smushed together and looks weird.

**Comment by nicholaszein** [comment]
Hey everyone!

Thank you for joining this discussion and sharing your feedback! 🙏

We can confirm Global Widgets are not showing correctly in some cases.

The issue has already been identified and our team is diligently working on a fix as we speak.

We hope to have a patch for this issue soon! 😉

We apologize for the disruption this may have caused to your workflow, and thank you for your patience as we work on a fix.


Kind regards

**Comment by nicholaszein** [comment]
Hi @StenderIternum,

we have made no changes to the Slides or Carousel widgets. This issue has no relation to the problem with Global Widgets as far as components go, unless you're using them as Global Widgets. 

You might want to check your cache or possible conflicting plugins.

@vinnco, **we do not recommend rolling back at the moment**, but if you want to roll back, you can roll back Elementor Core to version 3.24.7, and Pro to version 3.24.4.

**Comment by x1code** [comment]
Yep, fixed only for logged in users. More and more bad updates from Elementor...

**Comment by thisisbohau** [comment]
> Hi @StenderIternum,
> 
> we have made no changes to the Slides or Carousel widgets. This issue has no relation to the problem with Global Widgets as far as components go, unless you're using them as Global Widgets.
> 
> You might want to check your cache or possible conflicting plugins.
> 
> @vinnco, **we do not recommend rolling back at the moment**, but if you want to roll back, you can roll back Elementor Core to version 3.24.7, and Pro to version 3.24.4.

@nicholaszein even if you did not change anything inside the slider/carousel component it is still broken. Previously you could change the position of the navigation arrows and the pagination dots, now when you change the settings for the position NOTHING happens with the element. That also applies to elements created before the update.

**Comment by majus82** [comment]
I jump in, my case is in tier 3 at the moment. They reproduced it on a staging site of mine. I am happy I am not alone. Lots of broken Websites here. Thank you guys for looking into it. 
Yesterday's fix reverted the then persisting bug into not being missing on front-end but now in back-end and globals cannot be detached and edited anymore.

**Comment by matthewhandy1** [comment]
Same issue for me after updating to 3.25.1

**Comment by OGM444** [comment]
Tried rolling back to a previous version, got the forms working again but now experiencing a server error for all submissions

**Comment by nicholaszein** [comment]
Update:

We are running final test for the patch.

**Comment by davseve** [comment]
Hi, We have just released a fix for this is issue.
Please update to v3.25.2

**Comment by devdan26** [comment]
> Hi, We have just released a fix for this issue. Please update to v3.25.2

WILL IT BRING BACK THE GLOBAL WIDGETS THAT WERE REMOVED OR WE HAVE TO MANUALLY ADD THEM NOW, PLEASE REPLY! It's in the middle of the night here and my client's texting me with their site missing elements, goddamn.

**Comment by nicholaszein** [comment]
Hey everyone!

We have great news! 🙌 

### 📢 We're happy to announce that the issue with Global Widgets was solved with the release of Elementor Core v3.25.2! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

The widget should be restored to their original locations as long as you haven't edited the documents where they were located. 

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
